core/src/main/java/jenkins/model/item_category/ItemCategory.java

```diff
@@ -0,0 +1,90 @@
+package jenkins.model.item_category;
+
+import hudson.Extension;
+import hudson.ExtensionPoint;
+import hudson.model.ModelObject;
+import jenkins.model.Messages;
+
+/**
+ * A category for {@link hudson.model.Item}s.
+ */
+public abstract class ItemCategory implements ModelObject, ExtensionPoint {
+
+    public static int MIN_WEIGHT = Integer.MIN_VALUE;
+
+    public static int MIN_TOSHOW = 1;
+
+    /**
+     * Identifier, e.g. "category-id-default", etc.
+     *
+     * @return the identifier
+     */
+    public abstract String getId();
+
+    /**
+     * The icon class specification e.g. 'category-icon-folder', 'category-icon-default', etc.
+     *
+     * @return the icon class specification
+     */
+    public abstract String getIconClassName();
+
+    /**
+     * The description in plain text
+     *
+     * @return the description
+     */
+    public abstract String getDescription();
+
+    /**
+     * Helpful to set the order.
+     *
+     * @return the weight
+     */
+    public abstract int getWeight();
+
+    /**
+     * Minimum number required to show the category.
```

Comment:
@recena What I meant was whether there's a magic value (e.g. 0) that declares that "I don't care about this, do whatever makes sense on the UI".